cpp/src/arrow/util/config.in.cmake

```diff
@@ -0,0 +1,32 @@
+/*  Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+
+ This module sets the following variables in your project::
+
+   arrow_FOUND - true if arrow found on the system   */
+
+/* my_config.h file expanded by Cmake for build */
+
+
+#ifndef MY_CONFIG_H
+#define MY_CONFIG_H
+
+/* Support Armv8 CRC instructions  */
+#cmakedefine ARROW_USE_ARMCE 1
```

Comment:
This file shouldn't be required. Instead you can add preprocessor definitions directly in the cmake files, using e.g. the [add_definitions](https://cmake.org/cmake/help/v3.13/command/add_definitions.html?highlight=add_definition) command.

Reply:
Really thanks for comments.
'add_definitions' looks like a switch to enable/disable the macro by the command like 
"cmake 　-Dxxx＝on/off".
In my opinion, it's better to detect whether the macro is supported or not on cmake phase and dynamically  to define or not define the Macro related.  And it is also convenient and explicit to manage the macro definition.

Reply:
As the doc says, it "Adds definitions to the **compiler command line**" (emphasis mine). So you can use it to define a preprocessor symbol.